README.md

```diff
@@ -10,43 +10,261 @@
+http_asgi_app = get_asgi_application()
+
+from channels.routing import ProtocolTypeRouter, URLRouter
+
+application = ProtocolTypeRouter(
+    {
+        "http": http_asgi_app,
+        "websocket": URLRouter(
+          # add a path for IDOM's websocket
+          [idom_websocket_path()]
+        ),
+    }
+)
+```
+
+## `settings.py`
+
+In your settings you'll need to add `django_idom` to the
+[`INSTALLED_APPS`](https://docs.djangoproject.com/en/3.2/ref/settings/#std:setting-INSTALLED_APPS)
+list:
 
 ```python
-# Example code goes here
+INSTALLED_APPS = [
+  ...,
+  "django_idom",
+]
 ```
 
-For examples on how to use IDOM, [read the IDOM Documentation](https://idom-docs.herokuapp.com).
+You may configure additional options as well:
+
+```python
+# the base URL for all IDOM-releated resources
+IDOM_BASE_URL: str = "_idom/"
+
+# ignore these INSTALLED_APPS during component collection
+IDOM_IGNORE_INSTALLED_APPS: list[str] = ["some_app", "some_other_app"]
+```
+
+## `urls.py`
+
+You'll need to include IDOM's static web modules path using `idom_web_modules_path`.
+Similarly to the `idom_websocket_path()`, these resources will be used globally.
+
+```python
+from django_idom import idom_web_modules_path
+
+urlpatterns = [
+    idom_web_modules_path(),
+    ...
+]
+```
+
+## `sub_app/components.py`
+
+This is where, by a convention similar to that of
+[`views.py`](https://docs.djangoproject.com/en/3.2/topics/http/views/), you'll define
+your [IDOM](https://github.com/idom-team/idom) components. Ultimately though, you should
+feel free to organize your component modules you wish.
+
+```python
+import idom
+
+@idom.component
+def Hello(name):  # component names are camelcase by convention
+    return idom.html.h1(f"Hello {name}!")
+```
+
+## `sub_app/idom.py`
+
+This file is automatically discovered by `django-idom` when scanning the list of
+[`INSTALLED_APPS`](https://docs.djangoproject.com/en/3.2/ref/settings/#std:setting-INSTALLED_APPS).
+All apps that export components will contain this module.
+
+Inside this module must be a `components` list that is imported from
+[`components.py`](#sub_appcomponents.py):
+
+```python
+from .components import Hello
+
+components = [
+    Hello,
+    ...
+]
+```
+
+You may alternately reference the components with strings for the purpose of renaming:
+
+```python
+from .components import Hello as SomeOtherName
+
+components = [
+    "SomeOtherName",
+    ...
+]
+```
+
+## `sub_app/templates/your-template.html`
+
+In your templates, you may inject a view of an IDOM component into your templated HTML
+by using the `idom_view` template tag. This tag which requires the name of a component
+to render (of the form `app_name.ComponentName`) and keyword arguments you'd like to
+pass it from the template.
+
+```python
+idom_view app_name.ComponentName param_1="something" param_2="something-else"
+```
+
+In context this will look a bit like the following...
+
+```html
```

Comment:
change this to markdown codeblock from `html` to `jinja` for better syntax highlighting